File: c++/Solution100.cpp
//DFS O(min(V, U)) O(min(V, U)), V, U為兩tree之節點數
class Solution100 {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if(!p || !q) return !p && !q;
        if(p->val != q->val) return false;

        return isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
    }
};

File: java/Solution100.java
package leetCode.java;

//DFS O(min(V, U)) O(min(V, U)), V, U為兩tree之節點數
class Solution100 {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if (p == null || q == null) {
            return p == null && q == null;
        }
        if (p.val != q.val) {
            return false;
        }
        return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
    }
}